linchpin/__init__.py

```diff
@@ -688,6 +708,12 @@ def do_action(self, provision_data, action='up', run_id=None, tx_id=None):
 
             if provision_data[target].get('layout', None):
                 l_data = provision_data[target]['layout']
+                try:
+                    self._validate_layout(l_data)
```

Comment:
should this be not using the `_`?

Reply:
I accidentally branched off of my branch for #662 instead of develop.  Now that that's merged, I'll go back and clean things up